YAFC/Workspace/ProductionTable/ProductionTableFlatHierarchy.cs

```diff
@@ -82,12 +87,17 @@ public class FlatHierarchy<TRow, TGroup> where TRow : ModelObject<TGroup>, IGrou
             flatGroups.MoveListElementIndex(indexFrom, indexTo);
         }
 
-        //private readonly List<(Rect, SchemeColor)> listBackgrounds = new List<(Rect, SchemeColor)>();
         private readonly Stack<float> depthStart = new Stack<float>();
+
         private void SwapBgColor(ref SchemeColor color) {
-            color = color == SchemeColor.Background ? SchemeColor.PureBackground : SchemeColor.Background;
+            color = nextRowIsHighlighted ? nextRowBackgroundColor :
```

Comment:
I suggest to swap from ternaries to the usual `if` where possible to increase the readability. Especially in the cases where the ternary takes more than one line.

Reply:
This is what I did for the overproduction coloring, as it became too complex for ternaries at some point. So I agree it would be better to directly use `if` statements instead.

Reply:
Nice, this hybrid form works for me.

Only I would revert the logic:
```
if (nextRowIsHighlighted)
  color = nextRowBackgroundColor
else
  // Comment explainingt about alternating rows
  color = color == SchemeColor.Background ? SchemeColor.PureBackground : SchemeColor.Background;
```

Reply:
Would be more concise. Good find.

I refrained from using alternating colors for highlighted rows, since that would mean an additional bunch of color constants and enum values, leading to even more clutter.